source/adapters/native_cpu/enqueue.cpp

```diff
@@ -63,6 +63,23 @@ static native_cpu::state getResizedState(const native_cpu::NDRDescT &ndr,
 }
 #endif
 
+class LaunchInfoLocalArgs {
```

Comment:
Naming wise, it might be better to have something like `LaunchInfoGroup`, that's closer to what it is, I think. The fact that it calls `handleLocalArgs()` is less what this class is for and more just something that happens to be needed to be done for each group.